File: tests/bindings/dotnet/foo.Tests/StructureTest.cs
using System;
using Xunit;
using foo;

namespace foo.Tests
{
    class EmptyInterface : IEmptyInterface {}

    public class StructureTest
    {                       
        private static void CheckNumbersDefaults(Numbers x)
        {
            Assert.Equal(1u, x.Uint8Value);
            Assert.Equal(-1, x.Int8Value);
            Assert.Equal(2u, x.Uint16Value);
            Assert.Equal(-2, x.Int16Value);
            Assert.Equal(3u, x.Uint32Value);
            Assert.Equal(-3, x.Int32Value);
            Assert.Equal(4u, x.Uint64Value);
            Assert.Equal(-4, x.Int64Value);
            Assert.Equal(12.34f, x.FloatValue);
            Assert.Equal(-56.78, x.DoubleValue);
        }

        private static void CheckInnerStructureDefaults(InnerStructure x)
        {
            Assert.NotNull(x.InterfaceField);
            CheckNumbersDefaults(x.NumbersField);
        }

        private static void CheckStructureDefaults(Structure x)
        {
            Assert.False(x.BooleanFalse);
            Assert.True(x.BooleanTrue);
            Assert.Equal(StructureEnum.Var1, x.EnumVar1);
            Assert.Equal(StructureEnum.Var2, x.EnumVar2);
            Assert.Equal(TimeSpan.FromMilliseconds(4200), x.DurationMillis);
            Assert.Equal(TimeSpan.FromSeconds(76), x.DurationSeconds);
            Assert.Equal("Hello", x.StringHello);
            CheckInnerStructureDefaults(x.InnerStructure);
        }

        [Fact]
        public void StructureConstructorTest()
        {
            var x = new Structure(new InnerStructure(new EmptyInterface()));
            CheckStructureDefaults(x);
        }

       
       
    }
}

File: tests/bindings/java/foo-tests/src/test/java/io/stepfunc/foo_test/StructureTest.java
package io.stepfunc.foo_test;

import io.stepfunc.foo.*;
import org.junit.jupiter.api.Test;

import java.time.Duration;

import static org.assertj.core.api.Assertions.assertThat;
import static org.joou.Unsigned.*;

public class StructureTest {
    static class TestInterface implements EmptyInterface {}

     private static void checkNumbersDefaults(Numbers x)
     {
        assertThat(x.uint8Value).isEqualTo(ubyte(1));
        assertThat(x.int8Value).isEqualTo((byte)-1);
        assertThat(x.uint16Value).isEqualTo(ushort(2));
        assertThat(x.int16Value).isEqualTo((short)-2);
        assertThat(x.uint32Value).isEqualTo(uint(3));
        assertThat(x.int32Value).isEqualTo(-3);
        assertThat(x.uint64Value).isEqualTo(ulong(4));
        assertThat(x.int64Value).isEqualTo(-4);
        assertThat(x.floatValue).isEqualTo(12.34f);
        assertThat(x.doubleValue).isEqualTo(-56.78);
    }

    private static void checkInnerStructure(InnerStructure x) {
        assertThat(x.interfaceField).isNotNull();
        checkNumbersDefaults(x.numbersField);
    }

    private static void checkStructure(Structure x) {
        assertThat(x.booleanTrue).isTrue();
        assertThat(x.booleanFalse).isFalse();
        assertThat(x.enumVar1).isEqualTo(StructureEnum.VAR1);
        assertThat(x.enumVar2).isEqualTo(StructureEnum.VAR2);
        assertThat(x.durationMillis).isEqualTo(Duration.ofMillis(4200));
        assertThat(x.durationSeconds).isEqualTo(Duration.ofSeconds(76));
        assertThat(x.stringHello).isEqualTo("Hello");
        checkInnerStructure(x.innerStructure);
    }

    @Test
    public void testStructureConstructor() {
        Structure x = new Structure(new InnerStructure(new TestInterface()));
        checkStructure(x);
    }

    @Test
    public void testStructureBuilderMethods() {
        Structure x = new Structure(new InnerStructure(new TestInterface()))
            .withBooleanFalse(true) // introducing some chaos
            .withBooleanTrue(false);

        assertThat(x.booleanTrue).isFalse();
        assertThat(x.booleanFalse).isTrue();
    }
}
